### DISCUSSION TITLE
How to upload larger files to S3 using Bref and Lambda

### DISCUSSION THREAD

**Discussion by mslabba** [brefphp/bref#1089] [Support]
Hi,
I am using Bref for uploading my laravel application to AWS Lamba. Now the issue I am facing is I am unable to upload files more than 6MB to S3 in the application. Is there any option to increase the max upload size?

**Comment by mnapoli** [comment]
Hi, as explained in the docs you cannot upload more than 6MB to Lambda: https://bref.sh/docs/runtimes/http.html#binary-requests-and-responses

You'll have to upload directly to S3 (via presigned URLs) instead (no limit there).

**Reply by LemarinelNet** [reply]
Hi @mnapoli 
Tried to setup presigned url in my bref project : 
When the project is ran locally (on my own machine), the upload works (using my own aws access key)
But as soon I run it inside a lambda, the access key changes (starts with ASIA which refers to temporary keys), and the upload doesn't work anymore !

Any clue about how to setup presigned urls into bref ?

**Reply by LemarinelNet** [reply]
Just found the answer by myself: 
When instanciating the S3 Client, instead of setting credentials (Secret + key), nothing must be provided. It seems the S3 client will then use the "current" IAM role, which just works !
